src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -350,16 +354,41 @@ private void initializeFromClientConfig(JedisClientConfig config) {
         auth(credentialsProvider);
       }
 
-      int dbIndex = config.getDatabase();
-      if (dbIndex > 0) {
-        select(dbIndex);
-      }
-      String clientName = config.getClientName();
-      if (clientName != null) {
-        // TODO: need to figure out something without encoding
-        clientSetname(clientName);
-      }
+      try {
+        List<CommandArguments> fireAndForgetMsg = new ArrayList<>();
+
+        int dbIndex = config.getDatabase();
+        if (dbIndex > 0) {
+          fireAndForgetMsg.add(new CommandArguments(Command.SELECT).add(Protocol.toByteArray(dbIndex)));
+        }
+
+        String clientName = config.getClientName();
+        if (clientName != null) {
+          fireAndForgetMsg.add(new CommandArguments(Command.CLIENT).add(Keyword.SETNAME).add(clientName));
+        }
+
+        String libName = JedisMetaInfo.getArtifactId();
+        if (libName != null) {
+          fireAndForgetMsg.add(new CommandArguments(Command.CLIENT).add(Keyword.SETINFO)
+              .add(ClientAttrOption.LIBNAME.getRaw()).add(libName));
+        }
+
+        String libVersion = JedisMetaInfo.getVersion();
+        if (libVersion != null) {
+          fireAndForgetMsg.add(new CommandArguments(Command.CLIENT).add(Keyword.SETINFO)
+              .add(ClientAttrOption.LIBVER.getRaw()).add(libVersion));
+        }
+
+        for (CommandArguments arg : fireAndForgetMsg) {
+          sendCommand(arg);
+        }
 
+        // getMany and ignore result
+        getMany(fireAndForgetMsg.size());
+
+      } catch (JedisDataException e) {
+        // just ignore dataException, such as unknown command
```

Comment:
We can ignore unknown command but we should throw errors like illegal characters, etc.

Reply:
that's fine to me